crates/block-producer/src/mempool/mod.rs

```diff
@@ -110,6 +99,25 @@ pub struct Mempool {
 }
 
 impl Mempool {
+    /// Creates a new [Mempool] with the provided configuration.
+    pub fn new(
+        chain_tip: BlockNumber,
+        batch_limit: usize,
+        block_limit: usize,
+        state_retention: usize,
```

Comment:
I wonder if we should have something like `MempoolConfig` or `MempoolOptions` struct to encapsulate these. But this can also be done later on.